hs_core/views/resource_metadata_rest_api.py

```diff
@@ -0,0 +1,217 @@
+import logging
+
+from rest_framework.response import Response
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework import status
+from rest_framework import generics
+from rest_framework import serializers
+
+from hs_core import hydroshare
+from hs_core.models import Contributor, CoreMetaData, Coverage, Creator, Date, \
+    ExternalProfileLink, Format, FundingAgency, Identifier, Subject, Source, Relation
+from hs_core.views import utils as view_utils
+from hs_core.views.utils import ACTION_TO_AUTHORIZE
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalProfileLinkSerializer(serializers.Serializer):
```

Comment:
I have to do longhand anyway because many of the objects are erroring out as "Not JSON Serializable" and are nested serializers anyway. I'm open to hearing about the benefits though!